### PR TITLE
Allow copying extra folders to sandbox on deployment time

### LINKED ISSUE #509: Missing rimraf dependency in cli package
# Bug Report

## Current Behavior

When trying to use the latest `cli` package `v0.9.3` globally, the `rimraf` dependency is missing.

## Expected behavior

When using the `cli` globally installed it should execute the commands without throwing a missing dependency error. e.g., `boost deploy -e production`

## Possible Solution

Add `rimraf` into the `cli` package dependencies.

### Environment

- Booster version: `0.9.3`
- Node/npm version: 12.20.0
- OS: OSX 11.1 Big Sur

### LINKED ISSUE #580: Unexpected requirements on new deployment (0.10.0)
# Bug Report

## Current Behavior

With the new improved deployment process, I am getting new errors due to the following requirements:

- Missing file `.eslintignore`

```
 Error: ENOENT: no such file or directory, copyfile '.eslintignore' -> '.deploy/.eslintignore'
    Code: ENOENT
```

- Missing file `tsconfig.eslint.json`

```
Error: ENOENT: no such file or directory, copyfile 'tsconfig.eslint.json' -> '.deploy/tsconfig.eslint.json'
    Code: ENOENT
```

- Missing package `rimraf` (global install).

```
rimraf ./dist tsconfig.tsbuildinfo

sh: rimraf: command not found
npm ERR! code ELIFECYCLE
```

- `index.js` path is wrong.

```
 Error: Error: Cannot find module '/MyWonderfulBoosterProject/dist/index.js'
```

## Expected behavior

I expected a deployment without this new undocumented requirements.

## Possible Solution

* Optional copy of `.eslintignore` and `tsconfig.eslint.json` (only if exists).
* Add `rimraf` as a dependency of Booster and change call to local (without path inference).
* Change userApp directory:

Before:

``` typescript
const userApp = require(path.join(process.cwd(), 'dist', 'index.js'));
```

After:

``` typescript
const userApp = require(path.join(process.cwd(), '.deploy/dist', 'index.js'));
```

### Environment

- Booster version: [e.g. 0.8.2]  `0.10.0`
- Node/npm version: [e.g. Node 8/npm 5]  `node 14.8.0`/`npm 6.14.7`
- OS: [e.g. OSX 10.13.4, Windows 10] `MacOS Big Sur 11.1`

### PR DESCRIPTION
**Pull request by alvaroloes**
## Description
The main purpose of this PR is to allow the user to specify a set of assets or static resources that will be included in the deployed lambda's code.

Ideal to, for example, use .env files.

# Additional fixed included
- Changed the package.json template so that now the scripts are executed with `npx`
- Removed the files related to linter configuration when creating the deployment sandbox

Fix #509 
Fix #542 
Fix #580 

## Checks
- [x] Project Builds
- [x] Project passes tests and checks
- [x] Updated documentation accordingly

### CODE REVIEW

File: packages/framework-integration-tests/src/config/config.ts [hunk 1]
```diff
@@ -28,4 +29,5 @@ Booster.configure('production', (config: BoosterConfig): void => {
 
   config.appName = 'my-store-' + appNameSuffix
   config.provider = AWS.Provider()
+  config.staticResourceFolders = ['public']
```
Comment (alvaroloes) [inline packages/framework-integration-tests/src/config/config.ts:32]: The user configures an array of folders that will be zipped with the lambda code. Later they will be able to read it inside command and event handlers

File: packages/cli/src/common/sandbox.ts [hunk 2]
```diff
@@ -13,14 +13,24 @@ const copyFolder = (origin: string, destiny: string): void => {
   })
 }
 
-export const createSandboxProject = (sandboxPath: string): string => {
+export const createSandboxProject = (sandboxPath: string, assets?: Array<string>): string => {
   rmdirSync(sandboxPath, { recursive: true })
   mkdirSync(sandboxPath, { recursive: true })
   copyFolder('src', path.join(sandboxPath, 'src'))
 
-  const projectFiles = ['.eslintignore', 'package.json', 'tsconfig.eslint.json', 'tsconfig.json']
```
Comment (alvaroloes) [inline packages/cli/src/common/sandbox.ts:21]: These files are never needed when creating sandboxes

File: packages/framework-integration-tests/src/config/config.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ if (process.env.BOOSTER_ENV === 'local') {
 Booster.configure('development', (config: BoosterConfig): void => {
   config.appName = 'my-store'
   config.provider = AWS.Provider()
+  config.assets = ['assets', 'assetFile.txt']
```
Comment (alvaroloes) [inline packages/framework-integration-tests/src/config/config.ts:18]: The user configures an array of folders and files that will be zipped with the lambda code. Later they will be able to read it inside command and event handlers

**Review by NickSeagull — APPROVED** [review]
Thank. you a lot! 🙏

**Review by adriangmweb — APPROVED** [review]
LGTM! 🚀

**Pull request merged**